hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/OMKeyRenameResponse.java

```diff
@@ -73,13 +75,26 @@ public void addToDBBatch(OMMetadataManager omMetadataManager,
       BatchOperation batchOperation) throws IOException {
     String volumeName = renameKeyInfo.getVolumeName();
     String bucketName = renameKeyInfo.getBucketName();
+    String fromDbKey = omMetadataManager
+        .getOzoneKey(volumeName, bucketName, fromKeyName);
     omMetadataManager.getKeyTable(getBucketLayout())
-        .deleteWithBatch(batchOperation,
-            omMetadataManager.getOzoneKey(volumeName, bucketName, fromKeyName));
+        .deleteWithBatch(batchOperation, fromDbKey);
     omMetadataManager.getKeyTable(getBucketLayout())
         .putWithBatch(batchOperation,
             omMetadataManager.getOzoneKey(volumeName, bucketName, toKeyName),
             renameKeyInfo);
+
+    String renameDbKey = omMetadataManager.getRenameKey(
+        renameKeyInfo.getVolumeName(), renameKeyInfo.getBucketName(),
+        renameKeyInfo.getObjectID());
+    OmKeyRenameInfo omKeyRenameInfo = omMetadataManager.getRenamedKeyTable()
+        .get(renameDbKey);
+    if (omKeyRenameInfo == null) {
+      omKeyRenameInfo = new OmKeyRenameInfo(fromDbKey);
+      omMetadataManager.getRenamedKeyTable().putWithBatch(
+          batchOperation, renameDbKey, omKeyRenameInfo);
+    }
```

Comment:
This means we are actually only storing the Ozone key's original name correct?

Reply:
Yes correct, We only need the original key name. In between key renames are not necessary. But keeping this to store `OmKeyRenameInfo` instead single `String`, so we can expand on it further if needed.